.github/workflows/_test_te.yaml

```diff
@@ -0,0 +1,47 @@
+name: ~test TransformerEngine
+
+on:
+  # Manual trigger
+  workflow_dispatch:
+  # Called from another workflow
+  workflow_call:
+    inputs:
+      JAX_TE_IMAGE:
+        type: string
+        description: 'JAX-TE image build by NVIDIA/JAX-Toolbox'
+        required: true
+        default: 'ghcr.io/nvidia/jax-te:latest'
+
+jobs:
+  te-unit-tests:
+    runs-on: [self-hosted, compute, V100]
+    steps:
+      - name: Print environment variables
+        run: env
+
+      - name: Print GPU information
+        run: nvidia-smi
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Pull JAX-TE image
+        shell: bash -x -e {0}
+        run: |
+          docker pull ${{ inputs.JAX_TE_IMAGE }}
+
+      - name: Run JAX-TE unit tests w/ docker
+        shell: bash -x {0}
+        run: |
+          docker run -v /tmp/pytest-log:/tmp/pytest-log --gpus all ${{ inputs.JAX_TE_IMAGE }} sh -c "pip install pytest pytest-reportlog cuda-python && pytest --report-log=/tmp/pytest-log/report.jsonl /opt/transformer-engine/tests/jax"
```

Comment:
👍 Given the richness of the JSON output, this should make it easier now to add a markdown table of all the failed tests to the [job summary](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/) via a subsequent job step.

Reply:
I'd suggest using `docker run --name=<NAME>` to split the long command line into multiple lines for better readability.

Reply:
> 👍 Given the richness of the JSON output, this should make it easier now to add a markdown table of all the failed tests to the [job summary](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/) via a subsequent job step.

That's a pretty cool feature!


> I'd suggest using `docker run --name=<NAME>` to split the long command line into multiple lines for better readability.

Maybe I'm missing something, but `--name` just makes it easier to find your container when you `docker ps`. Is your suggestion just to break up the shell command for readability, so something like this?

```yaml
- name: Run JAX-TE unit tests w/ docker
        shell: docker run -v /tmp/pytest-log:/tmp/pytest-log --gpus all ${{ inputs.JAX_TE_IMAGE }} sh {0}
        run: |
          pip install pytest pytest-reportlog cuda-python
          pytest --report-log=/tmp/pytest-log/report.jsonl /opt/transformer-engine/tests/jax"
```

Reply:
Actually on second thought, my code above probably wouldn't work b/c `{0}` would be replaced by a filepath that doesn't exist within that container, so maybe something like:

```yaml
- name: Run JAX-TE unit tests w/ docker
  shell: docker run -v /tmp/pytest-log:/tmp/pytest-log --gpus all ${{ inputs.JAX_TE_IMAGE }} bash -x -c "$(cat {0})"
  run: |
    pip install pytest pytest-reportlog cuda-python
    pytest --report-log=/tmp/pytest-log/report.jsonl /opt/transformer-engine/tests/jax
```

Reply:
Very cool. I didn't realize we could use the `shell:` key to solve this problem.
But I wonder if "$(`cat {0}`)" could destroy certain features in the script, such as heredoc (`EOF...EOF`) and functions. I recall that GitLab has line-ending issues when defining functions in the script section.